curriculum-wrap-up/activity-think-pair-share-review.md

```diff
@@ -0,0 +1,119 @@
+
+- Vocabulary and definitions
+- Clear examples (code examples or examples of application)
+- Descriptions of the 2-4 _most_ _important_ details about the topic
+
+### !callout-info
+
+## "Enough Depth For You In Six Months"
+
+Have you ever started a long book, video game, or TV show, dropped it, and then tried to return to it after six months?
+
+<br/>
+
+Did you remember everything you needed to remember in order to get back into it? You might have wanted a review about the story, what was going on, who the characters are, and what was important. Our goal is to create a "cheat sheet" to help give these kinds of details to you, but about programming.
+
+### !end-callout
+
+## Preparation
+
+If the facilitator desires to do a warm-up activity, they should prepare the warm-up activity.
+
+Each group will make a set of review slides or documents. In the end, there should be one shared slide deck or document for the entire class that contains everyone's slides. Depending on the facilitator's preference, it may make sense to prepare empty slide decks for groups or the whole classroom.
+
+## Activity Instructions
+
+Follow directions to be sorted into small groups. All small groups have four sets of topics represented in the Core curriculum.
+
+Within small groups, read the listed topics together. Then, assign one set of topics to each member.
+
+Then, follow the phases Think, Pair, and Share:
+
+1. (~15 min.) Individually create initial review material for your assigned topics and lessons.
+1. (~20-35 min.) With a pair, develop the review material into a shareable slide deck.
+1. (Remaining time) With a large group, share and present the created review material.
+1. In the end, follow instructions to add your review slides to the classroom's collective review document.
+
+### Think Phase
+
+Consider your assigned set of topics. These topics are all names of topics in the curriculum.
+
+Begin to create a review guide for every lesson within those assigned topics. Write your notes on paper or in a document.
+
+Find:
+
+- important vocabulary and clear definitions
+- descriptions of important concepts
+- straightforward code examples
+- descriptions of the lesson's learning goals
+
+**It's probably impossible to document everything**.
+
+It's up to **you** to determine and prioritize **the most important information needed** for review. Use the full activity time available.
+
+### Pair Phase
+
+Pair with another person in your group.
+
+First, introduce your notes to each other.
+
+Then, collaboratively create a **maximum of four slides** for each set of topics. (Every person was assigned one set of topics. Therefore, a pair should have a maximum of eight slides created, and a group of four members should have a maximum of 16 slides.)
+
+Feel free to focus on code examples, diagrams, and definitions.
+
+### Share Phase
+
+Follow directions from the facilitator to share and present the material.
+
+## Groups and Topics
```

Comment:
How wedded are we to keeping topics that were chronologically close in the curriculum together here? For instance, Intro to Tests; While Loops; Exception Handling is a really weird grouping. It seems like While Loops would be better grouped with Iterating Over Data; Nested Data and Nested Loops. That leaves Representing Data which _could_ be grouped with Lists and Memory. Big O is a little wired to have in there, but it's not terrible (Lists and Memory talks about complexity of certain list operations). So then Intro to Tests and Exception Handling is its own group, which also isn't terrible, since tests talks about checking various conditions (including whether code raises and error) and Exceptions are related to conditions that we might want to test.

Then, later, grouping JS Tests and Algorithmic Strategies feels really odd. Algorithmic Strategies is distinct enough that I'd leave it on its own, and then maybe Tests can get paired with Functions? (Tests are implemented as Functions???)

Reply:
YAYYYYY I'm so excited for an extra brain on these groupings.

Hmmmmmmmm.

What do you think of (just highlighting relevant parts):

### Group 1

1. What is Software Development?; Intro to Dev Environment; Approaching a Problem
1. Intro to Debugging; Intermediate Python Debugging; **While Loops**
1. Intro to Functions; Functions, Continued
1. Representing Data; Iterating Over Data; Nested Data and Nested Loops

### Group 2

1. **Intro to Tests; Exception Handling**
1. Big O; Lists and Memory
1. Intro to Git; Branches in Git
1. Intro to OOP; Intro to OOP, Cont'd; Python OOP

### Group 5

1. Web Browsers; Structuring Content
1. Styles and Presentation; More Neat CSS Things; Layout
1. Learning Another Language
1. Functions; **Tests (Unit 3)**

### Group 6

1. **Algorithmic Strategies**
1. Adding Behavior; Event Handling; Calling APIs
1. React Components; Props
1. State and Event Handling; Lifting State Up; useEffect and Calling APIs

---

- Moving "While Loops" to the Debugging topic. This is definitely still treating "While loops" as a random topic still. I like this because I'm comfortable with a student spending three slides to summarize debugging and one slide to summarize while loops.
- Moving "Tests (Unit 3)" to "Functions." What do you think? To be honest, I'd rather have students spend all 4 slides covering regular functions, arrow functions, callback functions, and anonymous functions... rather than add in tests into the mix. And I'd rather have the Algorithmic Strategy students need to sacrifice a slide for Tests.

Reply:
Poor while loops. It just really wants to sit at the cool loops table, but nobody likes it. 😢 

I think this grouping makes good sense, with while loops being the notable outlier. WRT Algorithmic Strategies, I'm thinking more that whichever students end up with that one are going to have a hard enough time shifting their mindset to deal with that (it was a particularly heavy lesson) that dividing their thinking with another topic seems like it would undercut their thinking.

I'd still be really tempted to say that while loops could get grouped with the other looping topics. (1 slide for nested data structures, 1 slide for while loops, 1 slide for the for/in loop, 1 slide for dict helpers and range), but if that feels like too much, then I don't see a markedly better place to put it.

(As an aside, I'm seriously wondering why we even cover dynamic programming at all! I don't remember it being covered in my undergraduate CS program...)

Reply:
Haha! I'm on a mission to tie "representing data" and "nested data and loops" together.

Does adding While Loops onto this group feel okay? I'm feeling persuaded to bring the While Loops back home to family.

1. "Representing Data; Iterating Over Data; Nested Data and Nested Loops; While Loops"

I hear the Algo Strategy stress. Hmmmmm can I push us to taint the Algo Strategy group and protect the Function group this time around, and reassess how the activity goes afterwards?

(Regarding the aside: I am on Team _Let's really reconsider if we should cover dynamic programming_ myself.......)

Reply:
I think the loop family will appreciate the reunion!

Personally, I would still lean towards Tests being with Functions over Algo Strats (at least tests _are_ functions), but if you feel like that would detract from what folx will do with Functions, then AlgStrs is about the only place to drop it. Who knows. Maybe it will feel like whoever gets stuck with AS will at least have a clear partial win.